rotkehlchen/chain/ethereum/modules/convex/balances.py

```diff
@@ -17,14 +22,70 @@
     from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
 
 
+BalancesType = dict[ChecksumEvmAddress, dict[EvmToken, Balance]]
+
+
 class ConvexBalances(ProtocolWithBalance):
     """Query staked balances in the Convex gauges"""
 
     def __init__(self, database: DBHandler, evm_inquirer: 'EvmNodeInquirer'):
         super().__init__(database=database, evm_inquirer=evm_inquirer, counterparty=CPT_CONVEX)
+        self.cvx = A_CVX.resolve_to_evm_token()
+
+    def _query_staked_cvx(
+            self,
+            balances: BalancesType,
+            staking_contract: EvmContract,
+    ) -> None:
+        """
+        Query staking balances for CVX if there was a deposit event in Convex.
+        The logic is the same for locked cvx and staked cvx that is why staking_contract
+        is variable.
+        The balances variable is mutated in this function.
+        """
+        addresses_with_stake_mapping = self.addresses_with_deposits(
+            product=EvmProduct.CONVEX_STAKING,
+            deposit_events={(HistoryEventType.DEPOSIT, HistoryEventSubType.NONE)},
+        )
+        # addresses_with_deposits returns a mapping of address to evm event but since we will call
+        # the staking contracts with the addresses as arguments we need the list of addresses to
+        # index them later
+        addresses_with_stake = list(addresses_with_stake_mapping.keys())
+        if len(addresses_with_stake) == 0:
+            return None
+
+        cvx_price = Inquirer().find_usd_price(A_CVX)
+        call_output = self.evm_inquirer.multicall_2(
+            require_success=True,
+            calls=[(
+                staking_contract.address,
+                staking_contract.encode(method_name='balanceOf', arguments=[address]),
+            ) for address in addresses_with_stake],
+        )
 
-    def query_balances(self) -> dict[ChecksumEvmAddress, dict[EvmToken, Balance]]:
-        balances: dict[ChecksumEvmAddress, dict[EvmToken, Balance]] = {}
+        for idx, (status, result) in enumerate(call_output):
+            if status is False:
+                continue
+
+            address = addresses_with_stake[idx]
+            amount_raw = staking_contract.decode(result, 'balanceOf', arguments=[address])[0]  # pylint: disable=unsubscriptable-object  # noqa: E501
+            amount = asset_normalized_value(amount_raw, self.cvx)
+            if amount == ZERO:
+                continue
+
+            balance = Balance(amount=amount, usd_value=cvx_price * amount)
+            if address not in balances:
```

Comment:
These `not in` checks here and below look a bit ugly since this functionality can be achieved with default dicts. Can maybe the code be re-arranged a bit to use default dicts?

Reply:
Alexey suggested to use a `defauldict(lambda defauldict)` to avoid it. I just did that now